art/attacks/evasion/imperceptible_asr/imperceptible_asr.py

```diff
@@ -0,0 +1,621 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2020
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements the adversarial and imperceptible attack on automatic speech recognition systems of Qin et al.
+(2019). It generates an adversarial audio example.
+
+| Paper link: http://proceedings.mlr.press/v97/qin19a.html
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+from typing import TYPE_CHECKING, Tuple
+
+import numpy as np
+import scipy.signal as ss
+
+from art.attacks.attack import EvasionAttack
+from art.estimators.speech_recognition.speech_recognizer import SpeechRecognizerMixin
+from art.estimators.tensorflow import TensorFlowV2Estimator
+from art.utils import pad_audio_input
+
+if TYPE_CHECKING:
+    from tensorflow.compat.v1 import Tensor
+
+logger = logging.getLogger(__name__)
+
+
+class ImperceptibleAsr(EvasionAttack):
```

Comment:
It's Camel-case, but to follow the pattern of ImperceptibleASRPyTorch:
```suggestion
class ImperceptibleASR(EvasionAttack):
```